File: IS/front/src/ts/data/Color.ts
enum Color {
    RED = "RED",
    ORANGE = "ORANGE",
    WHITE = "WHITE",
    BROWN = "BROWN"
}

export const colorsToString = new Map<Color, string>()
colorsToString.set(Color.RED, "Красный")
colorsToString.set(Color.ORANGE, "Оранжевый")
colorsToString.set(Color.WHITE, "Белый")
colorsToString.set(Color.BROWN, "Коричневый")

export default Color

File: back/src/main/java/com/example/back/model/dto/Color.java
package com.example.back.model.dto;

public enum Color {
    RED,
    ORANGE,
    WHITE,
    BROWN;

    public static Color stringToColor(String s) {
        return switch (s.toLowerCase()) {
            case "red" -> RED;
            case "orange" -> ORANGE;
            case "white" -> WHITE;
            case "brown" -> BROWN;
            default -> null;
        };
    }
}